.ci/linux-steps.yaml

```diff
@@ -72,14 +76,18 @@ steps:
     cmake $CMAKEARGS -DPYTHON_EXECUTABLE=`which python` -DCEREAL_INCLUDE_DIR=/usr/include/ ..
   displayName: 'CMake'
 
-# Build mlpack
-- script: cd build && make -j2
+# Build mlpack (TODO: maybe remove VERBOSE=1)
+- script: cd build && VERBOSE=1 make -j2 
```

Comment:
So, I can see that ccache is being properly used in Azure Pipelines, but it doesn't seem to actually be caching anything across builds.  Do you want to approach that in this PR or just take the caching for Github Actions and call it a day?

I also found this: https://github.com/ryanpfalz/azure-pipelines-ccache-demo/blob/main/.azure-pipelines/ccache-pipeline.yml which is maybe useful here?  It looks like that includes a couple extra steps to save and restore the cache itself.

Reply:
I am out of patience for Azure Pipelines.  I gave it a day or two last week.  So I will "call it a day" and move on.